gno.land/pkg/keyscli/query_json.go

```diff
@@ -0,0 +1,84 @@
+package keyscli
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys/client"
+)
+
+func NewQueryJSONCmd(rootCfg *client.BaseCfg, io commands.IO) *commands.Command {
+	cfg := &client.QueryCfg{
+		RootCfg: rootCfg,
+	}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "jquery",
+			ShortUsage: "jquery [flags] <path>",
+			ShortHelp:  "EXPERIMENTAL: makes an ABCI query and return a result in json",
+			LongHelp:   "EXPERIMENTAL: makes an ABCI query and return a result in json",
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execQuery(cfg, args, io)
+		},
+	)
+}
+
+func execQuery(cfg *client.QueryCfg, args []string, io commands.IO) error {
+	if len(args) != 1 {
+		return flag.ErrHelp
+	}
+
+	cfg.Path = args[0]
+	if cfg.Path == "vm/qeval" {
+		// automatically add json suffix for qeval
+		cfg.Path = cfg.Path + "/json"
```

Comment:
looks weird. what's the goal? maybe add a comment if there is a valid reason.

Reply:
Will do. The reason for that is this `jquery` is made to handle JSON, so you want your query to return JSON. Therefore, you should not have to specify the format for the query. The goal is to automatically pass the argument to produce a JSON output without having to specify it every time.
`query  vm/qeval abc.xy/r/toto` -> machine
`jquery vm/qeval abc.xy/r/toto` -> JSON

Reply:
not valid anymore